### PR TITLE
chore(query): fix order for scalar cmp with null

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

By default, null is biggest than other values in databend. So we enable Scalar::Null compare with other none null scalars.

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

### CODE REVIEW

File: src/query/expression/src/values.rs [hunk 1]
```diff
@@ -770,6 +770,10 @@ impl PartialOrd for ScalarRef<'_> {
             (ScalarRef::Tuple(t1), ScalarRef::Tuple(t2)) => t1.partial_cmp(t2),
             (ScalarRef::Variant(v1), ScalarRef::Variant(v2)) => jsonb::compare(v1, v2).ok(),
             (ScalarRef::Geometry(g1), ScalarRef::Geometry(g2)) => compare_geometry(g1, g2),
+
+            // By default, null is biggest in pgsql
```
Comment (xudong963) [inline src/query/expression/src/values.rs:774]: If dialect isn't default(`Dialect::PostgreSQL`), should we change the behavior?

Comment (sundy-li) [reply src/query/expression/src/values.rs:774]: It’s hard to control by settings.

Comment (sundy-li) [reply src/query/expression/src/values.rs:774]: In order by clause, we can control the order by settings.

Here we add this rule to make min/max aggregation return stable results.

**Comment by sundy-li** [comment]
Bug in main for merge join, because it did not consider about null
```
drop table if exists t1;

drop table if exists t2;

create table t1(a int null);

insert into t1 values(1), (NULL), (2);

create table t2(b int null);

insert into t2 values(1), (NULL), (3);

select * from t1 join t2 on t1.a <= t2.b order by t1.a;
```

**Review by xudong963 — APPROVED** [review]

**Pull request merged**